File: 2022-05/2022-05-06/102. Binary Tree Level Order Traversal/solution.py
from typing import List, Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []

        stack = [root]
        output = []

        while len(stack) > 0:
            values = []
            n = len(stack)

            for _ in range(n):
                node = stack.pop(0)
                values.append(node.val)

                if node.left:
                    stack.append(node.left)

                if node.right:
                    stack.append(node.right)

            output.append(values)

        return output

File: 2022-05/2022-05-06/102. Binary Tree Level Order Traversal/solution.go
import "fmt"

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return nil
	}

	var output [][]int
	var stack = make([]*TreeNode, 0)

	stack = append(stack, root)
	fmt.Println(stack)

	for len(stack) > 0 {
		var sub_result []int
		stack_len := len(stack)
		for i := 0; i < stack_len; i++ {
			item := stack[i]
			sub_result = append(sub_result, item.Val)

			if item.Left != nil {
				stack = append(stack, item.Left)
			}
			if item.Right != nil {
				stack = append(stack, item.Right)
			}
		}

		stack = stack[stack_len:]
		output = append(output, sub_result)
	}

	return output
}